flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobServer.java

```diff
@@ -375,7 +378,32 @@ public File getFile(BlobKey key) throws IOException {
 	 * 		Thrown if the file retrieval failed.
 	 */
 	@Override
-	public File getFile(JobID jobId, BlobKey key) throws IOException {
+	public File getTransientFile(JobID jobId, BlobKey key) throws IOException {
+		checkNotNull(jobId);
```

Comment:
Should we check that `key` equals actually the `BlobType.TRANSIENT`?

Reply:
What about making the usage of blob keys failsafe by having different types of `BlobKey`. E.g. `TransientBlobKey` and `PermanentBlobKey`?